modules/patient-portal/src/main/java/io/uhndata/cards/patients/internal/SurveyFirstOpenedListener.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.uhndata.cards.patients.internal;
+
+import java.util.Calendar;
+import java.util.List;
+import java.util.Map;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.apache.sling.api.resource.observation.ResourceChange;
+import org.apache.sling.api.resource.observation.ResourceChangeListener;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.FieldOption;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.uhndata.cards.forms.api.FormUtils;
+import io.uhndata.cards.forms.api.QuestionnaireUtils;
+import io.uhndata.cards.links.api.Link;
+import io.uhndata.cards.links.api.LinkUtils;
+import io.uhndata.cards.resolverProvider.ThreadResourceResolverProvider;
+
+/**
+ * An {@link ResourceChangeListener} records when the patient proceeds to open the survey first time.
+ *
+ * @version $Id$
+ */
+@Component(immediate = true, property = {
+    ResourceChangeListener.PATHS + "=/Forms",
+    ResourceChangeListener.CHANGES + "=CHANGED"
+    })
+public class SurveyFirstOpenedListener implements ResourceChangeListener
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(SurveyFirstOpenedListener.class);
+
+    private static final String OPENED_PROP = "survey_opened";
+
+    private static final String BELONGS_TO_SURVEY_PROP = "belongsToSurvey";
```

Comment:
Not really a prop. Per slack discussion, let's rename to:
```suggestion
    private static final String LINK_DEFINITION_NAME = "belongsToSurvey";
```